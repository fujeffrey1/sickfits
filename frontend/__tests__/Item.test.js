import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'ABC123',
    title: 'A Cool Item',
    price: 4000,
    description: 'This item is really cool!',
    image: 'dog.jpg',
    largeImage: 'largedog.jpg'
};

// describe('<Item />', () => {
//     it('renders and displays properly', () => {
//         const wrapper = shallow(<ItemComponent item={fakeItem} />);
//         const PriceTag = wrapper.dive().find('.Item-price-3');
//         const Title = wrapper
//             .dive()
//             .find('.Item-color-4')
//             .at(1);
//         console.log(wrapper.dive().debug());
//         expect(PriceTag.children().text()).toBe('$50');
//         expect(Title.children().text()).toBe(fakeItem.title);
//     });
// });

describe('<Item />', () => {
    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        expect(toJSON(wrapper.dive())).toMatchSnapshot();
    });
});
