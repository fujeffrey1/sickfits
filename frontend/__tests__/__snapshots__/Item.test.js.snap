// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<WithStyles(Grid)
  item={true}
  sm={3}
>
  <WithStyles(Card)
    className="Item-card-1"
  >
    <WithStyles(CardActionArea)
      onClick={[Function]}
    >
      <WithStyles(CardMedia)
        className="Item-media-2"
        component="img"
        image="dog.jpg"
        title="A Cool Item"
      />
      <WithStyles(Typography)
        className="Item-price-3 Item-color-4"
        variant="h6"
      >
        $40
      </WithStyles(Typography)>
      <WithStyles(CardContent)>
        <WithStyles(Typography)
          className="Item-color-4"
          component="h2"
          gutterBottom={true}
          variant="h5"
        >
          A Cool Item
        </WithStyles(Typography)>
        <WithStyles(Typography)
          component="p"
        >
          This item is really cool!
        </WithStyles(Typography)>
      </WithStyles(CardContent)>
    </WithStyles(CardActionArea)>
    <WithStyles(CardActions)
      className="Item-actions-5"
    >
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
        size="small"
      >
        Edit
      </WithStyles(Button)>
      <AddToCart
        id="ABC123"
      />
      <DeleteItem
        id="ABC123"
      >
        Delete
      </DeleteItem>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</WithStyles(Grid)>
`;
